ui/src/App.vue

```diff
@@ -25,24 +27,39 @@ export default {
   },
   data() {
     return {
-      isRemote: {},
-      employmentTypes: {},
-      locations: {},
-      skills: {}
+      isRemote: [],
+      employmentTypes: [],
+      locations: [],
+      skills: []
```

Comment:
Will likely need to have some sort of state management, rather than just having all data at the top level like this.
Have a look at vuex (https://vuex.vuejs.org/) or Composition API (https://composition-api.vuejs.org/)

Reply:
@sanggonlee 
I created store file with vuex, but would it be weird after I declare state such as

export default new Vuex.Store({
    state: {
        isRemote: [],
        employmentTypes: [],
        locations: [],
        skills: []
    }
})

then use within GraphCard.vue instead of App.vue and assign value like this? (I will create a transform/mapper function to assign values)

mounted() {
    axios.get(mockAPI)
        .then(response => {
          this.$store.state.isRemote = response.data.is_remote.buckets;
          this.$store.state.employmentTypes = response.data.employment_type.buckets;
          this.$store.state.locations = response.data.location.buckets;
          this.$store.state.skills = response.data.skills.buckets;
        })
  }

Reply:
or would it be better use Getters within store file to assign values?

Reply:
> ```js
> export default new Vuex.Store({
>   state: {
>     isRemote: [],
>     employmentTypes: [],
>     locations: [],
>     skills: []
>   }
> })
> ```

This works, but I think it's better to keep the data as a whole, like this
```js
export default new Vuex.Store({
  state: {
    data: {},
  },
})
```

------------------

> ```js
> mounted() {
>   axios.get(mockAPI).then(response => {
>     this.$store.state.isRemote = response.data.is_remote.buckets;
>     this.$store.state.employmentTypes = response.data.employment_type.buckets;
>     this.$store.state.locations = response.data.location.buckets;
>     this.$store.state.skills = response.data.skills.buckets;
>   });
> }
> ```

Please use vuex actions and mutations instead of directly modifying the state, and then use getters to show the data.
```js
// component

methods() {
  ...mapActions(['setAnalytics']),
},
mounted() {
  axios.get(mockAPI).then(response => {
    this.setAnalytics(response.data);
  });
}
```
```js
// store

actions: {
  setAnalytics({ commit }, data) {
    commit(SET_ANALYTICS_DATA, data);
  },
},
mutations: {
 [SET_ANALYTICS_DATA](state, data) {
    state.data = data;
  },
}
```